### DISCUSSION TITLE
Handling Multi-Domain Routing in Next.js for a Marketplace

### DISCUSSION THREAD

**Discussion by adevinwild** [vercel/next.js#45215] [Help]
Hi

As we work on building our marketplace, we're thinking ahead to the **possibility of adding sub-domains and custom domains**. 
We want to make sure that **each seller can have their own "page" that searches for their products on the backend** (_keep in mind, we already have a separate server running on Node for all the business logic_). 

## Our question is 
**How can we use a single instance of Next.js to handle routing for multiple domains and manage parameters (like storeId) based on the domain?**

For example:

>shop.com -> **Show all products from all vendors**
>vendor01.com OR vendor01.shop.com -> **Show all products from vendor 01**

🚩 And especially how to manage SSL certificates...

Any suggestions or solutions would be greatly appreciated! 🤔💻

**Comment by wwwdepot** [comment]
> Learn how to programmatically create unique content pages for your users with a multi-tenant infrastructure using Edge Middleware.

For next.js part, you can use middleware, see [example ](https://github.com/vercel/examples/tree/main/edge-middleware/hostname-rewrites).

For hosting on Vercel, you open a ticket with their support.

**Comment by nabildroid** [comment]
use the middleware to extract the subdomain and rewrite the request to be something like this `[storeID]/....` where the store id is the subdomain
